lib/portlayer/portlayer.go

```diff
@@ -14,8 +14,41 @@
 
 package portlayer
 
-import "github.com/vmware/vic/lib/portlayer/storage"
+import (
+	log "github.com/Sirupsen/logrus"
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/vic/lib/metadata"
+	"github.com/vmware/vic/lib/portlayer/storage"
+	"github.com/vmware/vic/pkg/vsphere/extraconfig"
+	"github.com/vmware/vic/pkg/vsphere/session"
+	"golang.org/x/net/context"
+)
+
+var Config metadata.VirtualContainerHostConfigSpec
 
 type API interface {
 	storage.ImageStorer
 }
+
+func Init(ctx context.Context, sess *session.Session) error {
+	source, err := extraconfig.GuestInfoSource()
+	if err != nil {
+		return err
+	}
+
+	extraconfig.DecodeWithPrefix(source, &Config, "guestinfo.vch")
+	log.Debugf("decoded vch config: %#v", Config)
+	f := find.NewFinder(sess.Vim25(), false)
```

Comment:
https://github.com/vmware/vic/blob/master/pkg/vsphere/session/session.go#L85 is a finder instance but its 2nd argument is set to true (aka all = true). I was just suggesting adding another Finder (like FinderNoRecurse) into session and using it instead of creating one. I'm fine with both ways.